### PR TITLE
chore(cspi conditions): add cspi condition to represent the pool operations

### PR DESCRIPTION
**Pull request by mittachaitu**
This PR adds the conditions to represent pool operations in the status part
- Failure in Pool Import is represented in PoolLost condition
- PoolExpansion process is represented in PoolExpansion condition
- DiskReplacement process is represented in the BlockDeviceReplacement condition.
- DiskUnAvailable will be represented when underlying disk is detached/faulted.
- Events will be generated when the underlying disk is in other than the **ONLINE** state.
- It also updates the command from `zpool get compression <pool_name>` to `zfs get compression <pool_name>` because compression property is not supported via zpool command line.

**Sample Output When Operation Is in Progress**:
```yaml
  status:
    capacity:
      free: 19900M
      total: 19900M
      used: 92k
    conditions:
    - lastTransitionTime: "2020-04-10T03:44:42Z"
      lastUpdateTime: "2020-04-10T03:44:42Z"
      message: 'Pool expansion is in progress because of blockdevice/raid group addition
        error: <nil>'
      reason: PoolExpansionInProgress
      status: "True"
      type: PoolExpansionInProgress
    phase: ONLINE
    readOnly: false
```

**Sample Current Conditions Output**:
```yaml
  status:
    capacity:
      free: 19900M
      total: 19900M
      used: 95k
    conditions:
    - lastTransitionTime: "2020-04-10T03:56:47Z"
      lastUpdateTime: "2020-04-10T03:56:47Z"
      message: Blockdevice replacement was successfully completed
      reason: BlockDeviceReplacementSucceess
      status: "False"
      type: DiskReplacement
    - lastTransitionTime: "2020-04-10T03:56:57Z"
      lastUpdateTime: "2020-04-10T03:56:57Z"
      message: Pool expansion was successfull by adding blockdevices/raid groups
      reason: PoolExpansionSuccessful
      status: "False"
      type: PoolExpansionInProgress
    - lastTransitionTime: "2020-04-10T04:01:25Z"
      lastUpdateTime: "2020-04-10T04:01:56Z"
      message: failed to importcstor-1234pool
      reason: PoolLost
      status: "True"
      type: PoolLost
    - lastTransitionTime: "2020-04-14T10:28:54Z"
      lastUpdateTime: "2020-04-14T10:28:54Z"
      message: Following disks [sparse-177b6bc2ae2dd332c7a384a02179368b] are unavailable/faulted
      reason: DisksAreUnavailable
      status: "True"
      type: DiskUnavialable
    phase: ONLINE
    readOnly: false
```
Once the disk is available then DiskUnavailable condition will be updated to 
```yaml
    - lastTransitionTime: "2020-04-14T10:35:31Z"
      lastUpdateTime: "2020-04-14T10:35:31Z"
      reason: DisksAreAvailable
      status: "False"
      type: DiskUnavialable
```
**Sample events on corresponding CSPI when the disk was detached:**

```sh
Events:
  Type     Reason         Age                 From               Message
  ----     ------         ----                ----               -------
  Normal   Created        6m16s               CStorPoolInstance  Pool created successfully
  Normal   Pool Imported  5m (x2 over 6m)     CStorPoolInstance  Pool Import successful: cstor-1234
  Warning  DeviceState    4m1s (x10 over 6m)  CStorPoolInstance  sparse-177b6bc2ae2dd332c7a384a02179368b device was in UNAVAILABLE state
```

**Special Note To Reviewers**:
TODO: 
- Integration test PRs will be raised subsequently.

Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

### CODE REVIEW

File: pkg/controllers/cspi-controller/status.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	cspiutil "github.com/openebs/cstor-operators/pkg/controllers/cspi-controller/util"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+//TODO: Update the code to use patch instead of Update call
+
+// UpdateStatusConditionEventually updates the CSPI in etcd with provided
+// condition. Below function reties for three times to update the CSPI with
```
Comment (shubham14bajpai) [inline pkg/controllers/cspi-controller/status.go:32]: nit: retries

File: pkg/controllers/cspi-controller/handler.go [hunk 2]
```diff
@@ -189,11 +199,11 @@ func (c *CStorPoolInstanceController) update(cspi *cstor.CStorPoolInstance) (*cs
...
+	ncspi, err := oc.Update(cspi)
 	if err != nil {
-		return cspi, errors.Errorf("Failed to update pool due to %s", err.Error())
+		return ncspi, errors.Errorf("Failed to update pool due to %s", err.Error())
 	}
-	return c.updateStatus(cspi)
+	return c.updateStatus(ncspi)
 }
 
 func (c *CStorPoolInstanceController) updateStatus(cspi *cstor.CStorPoolInstance) (*cstor.CStorPoolInstance, error) {
```
Comment (shubham14bajpai) [inline pkg/controllers/cspi-controller/handler.go:209]: Can we rename this to `updateStatusCondition`

Comment (mittachaitu) [reply pkg/controllers/cspi-controller/handler.go:209]: I think it should be another way here we are really updating the entier status... func name in status.go will be updated to `updateStatusCondition`.

Comment (shubham14bajpai) [reply pkg/controllers/cspi-controller/handler.go:209]: Correct `UpdateStatus` needs to be renamed.

**Comment by sonasingh46** [comment]
Reason and message in status conditions will go to a proper place in subsequent PRs.

File: pkg/controllers/cspi-controller/status.go (hunk 1 above)
Comment (mittachaitu) [reply pkg/controllers/cspi-controller/status.go]: Done

File: pkg/controllers/cspi-controller/handler.go (hunk 2 above)
Comment (mittachaitu) [reply pkg/controllers/cspi-controller/handler.go:209]: Done

**Pull request merged**